monai/networks/nets/efficientnet.py

```diff
@@ -0,0 +1,805 @@
+        width_coefficient: float = 1.0,
+        depth_coefficient: float = 1.0,
+        dropout_rate: float = 0.2,
+        image_size: int = 224,
+        batch_norm_momentum: float = 0.99,
+        batch_norm_epsilon: float = 1e-3,
+        drop_connect_rate: float = 0.2,
+        depth_divisor=8,
+    ) -> None:
+        """
+        EfficientNet based on `Rethinking Model Scaling for Convolutional Neural Networks <https://arxiv.org/pdf/1905.11946.pdf>`_.
+        Adapted from `EfficientNet-PyTorch
+        <https://github.com/lukemelas/EfficientNet-PyTorch>`_.
+
+        Args:
+            blocks_args_str: block definitions.
+            spatial_dims: number of spatial dimensions.
+            in_channels: number of input channels.
+            num_classes: number of output classes.
+            width_coefficient: width multiplier coefficient (w in paper).
+            depth_coefficient: depth multiplier coefficient (d in paper).
+            dropout_rate: dropout rate for dropout layers.
+            image_size: input image resolution.
+            batch_norm_momentum: momentum for batch norm.
+            batch_norm_epsilon: epsilon for batch norm.
+            drop_connect_rate: dropconnect rate for drop connection (individual weights) layers.
+            depth_divisor: depth divisor for channel rounding.
+
+        Examples::
+
+            # for pretrained spatial 2D ImageNet
+            >>> image_size = get_efficientnet_image_size("efficientnet-b0")
+            >>> inputs = torch.rand(1, 3, image_size, image_size)
+            >>> model = EfficientNetBN("efficientnet-b0", pretrained=True)
+            >>> model.eval()
+            >>> outputs = model(inputs)
+
+            # create spatial 2D
+            >>> model = EfficientNetBN("efficientnet-b0", spatial_dims=2)
+
+            # create spatial 3D
+            >>> model = EfficientNetBN("efficientnet-b0", spatial_dims=3)
+
+            # create EfficientNetB7 for spatial 2D
+            >>> model = EfficientNetBN("efficientnet-b7", spatial_dims=2)
+
+        """
+        super().__init__()
+
+        if spatial_dims not in (1, 2, 3):
+            raise AssertionError("spatial_dims can only be 1, 2 or 3.")
+
+        # select the type of N-Dimensional layers to use
+        # these are based on spatial dims and selected from MONAI factories
+        conv_type: Type[Union[nn.Conv1d, nn.Conv2d, nn.Conv3d]] = Conv["conv", spatial_dims]
+        batchnorm_type: Type[Union[nn.BatchNorm1d, nn.BatchNorm2d, nn.BatchNorm3d]] = Norm["batch", spatial_dims]
+        adaptivepool_type: Type[Union[nn.AdaptiveAvgPool1d, nn.AdaptiveAvgPool2d, nn.AdaptiveAvgPool3d]] = Pool[
+            "adaptiveavg", spatial_dims
+        ]
+
+        # decode blocks args into arguments for MBConvBlock
+        blocks_args = _decode_block_list(blocks_args_str)
+
+        # checks for successful decoding of blocks_args_str
+        assert isinstance(blocks_args, list), "blocks_args should be a list"
```

Comment:
the project's style is to only use `assert` in the test files,
so, this could be (please revise the other few places)
```py
if not isinstance(blocks_args, list):
    raise ValueError...
```

Reply:
Thanks, I will resolve this in the next commit.